### PR TITLE
Features/filter recipes

### PR DESCRIPTION
**Pull request by laytonmaes**
# Description
Added filter by tag functionality and search by name functionality.

Addresses # (issue)
#24 #16 
we are not currently allowing for edge cases Ie. case sensitivity.

## Type of change
functional
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

we have tested functionality in the dom in chrome.

- [X] Dom

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -29,8 +37,24 @@ recipeSection.addEventListener('click', (e) => {
   displayRecipeDetail(e)
   })
 
+  filterBreakfast.addEventListener("click", () => {
```
Comment (angelewilliams) [inline src/scripts.js:40]: Glad this is working, will be interesting to refactor later on

File: src/classes/RecipeRepository.js [hunk 2]
```diff
@@ -34,6 +34,9 @@ class RecipeRepository {
 
   filterByName(recipeName){
     return this.recipeObjects.filter((recipeObject) => {
+      //input make lowercase
```
Comment (angelewilliams) [inline src/classes/RecipeRepository.js:37]: Nice, think this will be a good reminder to address edge cases

**Review by angelewilliams — APPROVED** [review]

File: src/scripts.js [hunk 3]
```diff
@@ -29,8 +37,24 @@ recipeSection.addEventListener('click', (e) => {
...
+  displayFilteredTags("lunch")})
+
+  filterDinner.addEventListener("click", () => {
+    displayFilteredTags("dinner")})
+
+  filterSnack.addEventListener("click", () => {
+      displayFilteredTags("snack")})
+
+  filterDip.addEventListener("click", () => {
+     displayFilteredTags("dip")})
```
Comment (michaelputnam67) [inline src/scripts.js:53]: could we do this with event bubble? #6

**Review by michaelputnam67 — COMMENTED** [review]
Love that we are pushing ahead without getting lost in rabbit holes. Great Job team!

**Pull request merged**